corehq/apps/auditcare/utils/migration.py

```diff
@@ -0,0 +1,61 @@
+from corehq.apps.auditcare.models import AuditcareMigrationMeta
+from django.core.cache import cache
+import gevent
+from dimagi.utils.dates import force_to_datetime
+
+
+class AuditCareMigrationUtil():
+
+    def __init__(self):
+        self.start_key = "auditcare_migration_2021_next_batch_time"
+        self.start_lock_key = f"{self.start_key}_lock"
+
+    def get_next_batch_start(self, counter=0):
+        if self.is_start_key_lock_acquired():
+            gevent.sleep(1)
+            if counter >= 10:
+                raise Exception("Unable to get next batch start time")
+            return self.get_next_batch_start(counter=counter + 1)
+        return cache.get(self.start_key)
+
+    def set_next_batch_start(self, value):
+        cache.set(self.start_key, value)
+
+    def acquire_read_lock(self):
+        cache.set(self.start_lock_key, 1)
```

Comment:
I assume the pattern for using this is something like
```py
util = AuditCareMigrationUtil()
if not util.is_start_key_lock_acquired():
    util.acquire_read_lock()
...
```

Which is vulnerable to a race condition. Like if two processes try to acquire the lock at exactly the same time they might both end up thinking they successfully acquired the lock. I think you want to look into the redis lock primitive rather than trying to roll your own lock with a flag.

Reply:
I will check out redis lock.
Thanks for the suggestion.

Reply:
Can this be removed now that you have switched to a redis lock?